src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -418,6 +424,43 @@ protected override void Free()
             }
         }
 
+        internal sealed class IsSimpleMethodVisitor : BoundTreeWalkerWithStackGuard
+        {
+            private bool _hasComplexity;
+
+            internal static bool IsSimpleMethod(BoundNode? node)
+            {
+                if (node is BoundMethodBodyBase methodBody)
+                {
+                    node = methodBody.BlockBody ?? methodBody.ExpressionBody;
```

Comment:
>node = methodBody.BlockBody ?? methodBody.ExpressionBody; [](start = 20, length = 57)

It feels like we should check both, we are visiting both if present in AbstractFlowPass.VisitMethodBodies.
For BoundConstructorMethodBody we also visit initializer.